src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1447,27 +1461,97 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
 
                         if (operandComparedToNull.Type?.IsReferenceType == true)
                         {
-                            int slot = MakeSlot(operandComparedToNull);
-
-                            if (slot > 0)
+                            var slotBuilder = ArrayBuilder<int>.GetInstance();
+                            getOperandSlots(operandComparedToNull, slotBuilder);
+                            Normalize(ref this.State);
```

Comment:
Can we move the `Normalize` call  down into the conditional (only normalize if slots were returned/created)?

Reply:
I just cannot win: https://github.com/dotnet/roslyn/pull/28722#discussion_r203945263

Reply:
I think @jcouv is just suggesting moving `Normalize` down two lines, into the `if`. Seems reasonable.

---
In reply to: [204224457](https://github.com/dotnet/roslyn/pull/28722#discussion_r204224457) [](ancestors = 204224457)

Reply:
Yes, that's what I meant.